geonode_mapstore_client/client/js/epics/index.js

```diff
@@ -109,6 +111,10 @@ export const gnSetDatasetsPermissions = (actions$, { getState = () => {}} = {})
                         return Rx.Observable.of(...stateLayers.map((l) => updateNode(l.id, 'layer', {perms: l.perms || []}) ));
                     });
             }
+
+            // skip layers of non-geonode origin
+            if (!action.layer?.pk) return Rx.Observable.empty();
```

Comment:
shouldn't we look for `extendedParams?.pk` instead as we doing in the previous check?

Reply:
when adding the layer, the layer object, holds the pk in both the root and extendedParams, unlike the one's in mapLayers. Nonetheless, I have updated it to maintain consistency